app/src/main/java/com/islaharper/dawnofandroid/di/UseCaseModule.kt

```diff
@@ -13,6 +17,30 @@ import javax.inject.Singleton
 @InstallIn(SingletonComponent::class)
 object UseCaseModule {
 
+    @Provides
+    @Singleton
+    fun provideSaveDarkModeUseCase(repository: Repository): SaveDarkModeUseCase {
```

Comment:
You can make these `@Inject` constructor:

```kotlin
class SaveDarkModeUseCase @Inject constructor(private val repository: Repository) {
    suspend operator fun invoke(isDarkMode: Boolean) {
        repository.saveDarkModeState(isDarkMode = isDarkMode)
    }
}
```

then you don't need to write the providers manually. This saves some boilerplate and the code generated by Dagger through the `@Inject` annotation is explained as more efficient.

Reply:
Thanks, noted and made the change for the next commit.